File: src/main/java/com/nugurang/graphql/mutation/TaskMutation.java
package com.nugurang.graphql.mutation;

import com.nugurang.dto.TaskDto;
import com.nugurang.dto.TaskInputDto;
import com.nugurang.service.TaskService;
import graphql.kickstart.tools.GraphQLMutationResolver;
import org.springframework.stereotype.Service;

@Service
public class TaskMutation implements GraphQLMutationResolver {
    private final TaskService taskService;

    public TaskDto createTask(TaskInputDto taskInputDto, Long workId) {
        return taskService.createTask(taskInputDto, workId).toDto();
    }

    public TaskDto updateTask(TaskInputDto taskInputDto, Long taskId) {
        return taskService.updateTask(taskInputDto, taskId).toDto();
    }

    public Long deleteTask(Long taskId) {
        taskService.deleteTask(taskId);
        return taskId;
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public TaskMutation(final TaskService taskService) {
        this.taskService = taskService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/mutation/TaskMutation.kt
package com.nugurang.graphql.mutation

import com.nugurang.dto.TaskDto
import com.nugurang.dto.TaskInputDto
import com.nugurang.mapper.TaskMapper
import com.nugurang.service.TaskService
import graphql.kickstart.tools.GraphQLMutationResolver
import org.springframework.stereotype.Service

@Service
class TaskMutation(
    private val taskService: TaskService,
    private val taskMapper: TaskMapper
) : GraphQLMutationResolver {

    fun createTask(taskInputDto: TaskInputDto, workId: Long): TaskDto {
        return taskService
            .createTask(taskInputDto, workId)
            .let(taskMapper::toDto)
    }

    fun updateTask(taskInputDto: TaskInputDto, taskId: Long): TaskDto {
        return taskService
            .updateTask(taskInputDto, taskId)
            .let(taskMapper::toDto)
    }

    fun deleteTask(taskId: Long): Long {
        taskService.deleteTask(taskId)
        return taskId
    }
}